website/project/model.py

```diff
@@ -1004,6 +1004,11 @@ def ids_above(self):
         return {p._id for p in parents}
 
     @property
+    def has_children(self):
+        if not self.nodes:
```

Comment:
This doesn't look to return anything in the true case and is also very negative. If there's something fancier going on here, just add a comment describing it.